lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -2068,7 +2068,7 @@ public function walkConditionalExpression($condExpr)
     /**
      * Walks down a ConditionalTerm AST node, thereby generating the appropriate SQL.
      *
-     * @param AST\ConditionalTerm $condTerm
+     * @param AST\ConditionalTerm|AST\ConditionalFactor|AST\ConditionalPrimary $condTerm
```

Comment:
I've tried using the interface here instead, but that fails, presumably because Psalm does not manage to compute that that interface minus `AST\ConditionalTerm` is `AST\ConditionalFactor|AST\ConditionalPrimary`